osu.Desktop/Program.cs

```diff
@@ -168,12 +169,20 @@ private static bool trySendIPCMessage(IIpcHost host, string cwd, string[] args)
 
         private static void setupVelopack()
         {
-            VelopackApp
-                .Build()
-                .WithFirstRun(v =>
-                {
-                    if (OperatingSystem.IsWindows()) WindowsAssociationManager.InstallAssociations();
-                }).Run();
+            var app = VelopackApp.Build();
+
+            if (OperatingSystem.IsWindows())
+                configureWindows(app);
+
+            app.Run();
+        }
+
+        [SupportedOSPlatform("windows")]
+        private static void configureWindows(VelopackApp app)
+        {
+            app.WithFirstRun(_ => WindowsAssociationManager.InstallAssociations());
+            app.WithAfterUpdateFastCallback(_ => WindowsAssociationManager.UpdateAssociations());
```

Comment:
Shouldn't this be enough to guarantee a fix after a single release? I don't understand the concern in the PR description regarding "two releases".

Reply:
I can't say for certain, which is why I err on the side of caution of keeping the logic for 2 releases. It shouldn't have much overhead, just becomes useless after a certain number of releases.

This is mostly up to your preference. I'd suggest doing a release and seeing if it fixes the issue. If so, then remove it next release. If not, keep it for another, then remove it after that.

Reply:
> Regardless, you can always just release and evaluate. I'd suggest doing a release and seeing if it fixes the issue for the majority of players. If so, then remove it in the subsequent release. If not, keep it for another, then remove it after that.

I'd hope this would be **tested before release and guaranteed to work**. Can I assume you haven't tested and that is on me to do?

Reply:
I have not yet tested it. But I'm willing to do that right now if needed.

Though, it'd take me a bit of time to get my environment set up (need to create a windows vm, build, transfer files, create github pat & repo, etc). I'd be looking at maybe an hour to get an environment ready.

Depending on how fast you want this out, you may test it yourself, or leave it for me. Let me know what you'd prefer.

Reply:
Put it this way: I wouldn't PR anything without testing myself. Do whatever you want with that information.

Reply:
Noted, will test it ASAP. Sorry for the difficulties, I'll let you know when I've made sure this works as intended.

Reply:
Post-Installation:
![image](https://github.com/user-attachments/assets/7016025a-5b74-4b2c-b718-838efbdf2289)

Pre-Update (modified to have pippi as part of path):
![image](https://github.com/user-attachments/assets/8d91bd8f-d848-44b9-9cd8-f117c2a3617d)

Post-Update:
![image](https://github.com/user-attachments/assets/d9423c33-e756-4dc9-8f59-7ef50c3998af)

Post-Uninstallation:
![image](https://github.com/user-attachments/assets/b65e4437-02c0-4aa5-99c4-a715167df936)

All of my tests pass on Windows 11 x64. It should only require one release as per my testing. Everything should be good to go.

Reply:
Thanks for testing 🙏